src/tls13.c

```diff
@@ -9367,14 +9801,48 @@ static int DoTls13CertificateVerify(WOLFSSL* ssl, byte* input,
                 ERROR_OUT(SIG_VERIFY_E, exit_dcv);
             }
 
-            sig->buffer = (byte*)XMALLOC(args->sz, ssl->heap,
+            sig->length = args->sz;
+#ifdef WOLFSSL_DUAL_ALG_CERTS
+            if (!wolfSSL_is_server(ssl) &&
+                ssl->sigSpec != NULL &&
+                *ssl->sigSpec == WOLFSSL_CKS_SIGSPEC_BOTH) {
+                /* If its RSA, we only hybridize with RSA3072 which has a sig
+                 * size of 384. For ECC, this is actually encoded as an RFC5912
+                 * formatted signature which means we can use the ASN APIs to
+                 * figure out the length. Note that some post-quantum sig algs
+                 * have variable length signatures (Falcon). That is why we
+                 * don't do:
+                 *   sig->length -= args->altSignatureSz; */
+
+                if (args->sigAlgo == rsa_pss_sa_algo) {
+                    sig->length = 384;
+                }
+                else if (args->sigAlgo == ecc_dsa_sa_algo) {
+                    word32 tmpIdx = args->idx;
+                    sig->length = wc_SignatureGetSize(WC_SIGNATURE_TYPE_ECC,
+                                      ssl->peerEccDsaKey,
+                                      sizeof(*ssl->peerEccDsaKey));
+                    if (GetSequence(input, &tmpIdx, (int*)&sig->length,
+                                    args->sz) < 0) {
+                        ERROR_OUT(SIG_VERIFY_E, exit_dcv);
+                    }
+                    /* We have to increment by the size of the header. */
+                    sig->length += tmpIdx - args->idx;
```

Comment:
Does this need an overflow / underflow check?

Reply:
I don't think so.  If it was going to be a crazy value, then GetSequence() would error out and we wouldn't get here.